### DISCUSSION TITLE
API and Countries, Regions and Cities.

### DISCUSSION THREAD

**Discussion by rossdav70** [plausible/analytics#3152] [Anything Else]
In my Plausible reports I see friendly Country , region and city names.  Via the API I get countries as just GB, US etc, Regions look like GB-ENG, GB-SCT etc and my cities are all just numbers. 

Is there a lookup I can use to 'translate' these to the same friendly terms I am seeing in the front end?

Thanks!

**Comment by ruslandoga** [comment]
👋 @rossdav70 

The numbers you are seeing are probably geoname ids: https://www.geonames.org/10237388/everest-base-camp.html

**Reply by ruslandoga** [reply]
Plausible uses https://github.com/plausible/location library for geoname id <-> city name mapping when rendering dashboards.

**Reply by rossdav70** [reply]
Thanks for this. I wonder if the API could be changed in the future to return the friendly versions?

**Reply by HenkVanMaanen** [reply]
I dont think its a good idea to return friendly versions because then you would also need to provide translations and other localization stuff. With the current option the client can decide for themselves how to show the friendly name. Also charts mostly work with ISO country codes.

**Reply by rossdav70** [reply]
@HenkVanMaanen yes fair enough. My problem is the use of API is through a Google Data studio connector that I have written for myself which just shows what it gets. I really want to be able to do some form of translation on it. I've switched from using Google Analytics and my clients will be expecting their reports to contain certain important information. City is quite a big deal for me unfortunately and at the moment I just get a number.
